apps/settings/js/modules/bluetooth/bluetooth_context.js

```diff
@@ -797,12 +802,20 @@ define(function(require) {
         return Promise.reject('default adapter is not existed!!');
       }
 
-      return this._defaultAdapter.pair(address).then(() => {
-        Debug('pair(): Resolved with void value');
+      // Note on Bluedroid stack, discovery has to be stopped before pairing 
+      // (i.e., call stopDiscovery() before pair()) otherwise stack callbacks 
+      // with pairing failure.
+      return this.stopDiscovery().then(() => {
+        return this._defaultAdapter.pair(address).then(() => {
+          Debug('pair(): Resolved with void value');
+        }, (reason) => {
+          Debug('pair(): Reject with this reason: ' + reason);
+          return Promise.reject(reason);
+        });  
       }, (reason) => {
-        Debug('pair(): Reject with this reason: ' + reason);
+        Debug('pair(): Reject in stopDiscovery with this reason: ' + reason);
```

Comment:
This is not required as in `this.stopDiscovery` you already printed the error message.